fix: responsive navbar

MiguelMRojas opened the pull request:
## Navbar Responsive






https://github.com/hawks-atlanta/frontend-react/assets/115849391/8bac936c-7687-496d-9d71-f5876c0ccd10

MiguelMRojas committed:
fix: responsive navbar

PedroChaparro reviewed (changes requested):
Try to make the responsive navbar to take the full width and add more spacing between the buttons. It can be hard for some users to click the right button.

MiguelMRojas committed:
fix: navbar width

MiguelMRojas committed:
style: format file

MiguelMRojas committed:
fix: navbar responsive

MiguelMRojas committed:
ci: test home and navbar

MiguelMRojas committed:
fix: navbar menu test

The selector has been adjusted to select the button by its aria-label attribute, which resolves the selection issue

PedroChaparro commented on src/components/Nav/NavbarButtonGroup.tsx:
```diff
@@ -0,0 +1,20 @@
+      >
+        {text}
+      </Link>
+      <span className="mx-1"></span>
```
Don't add an element just to add a margin.

PedroChaparro commented on src/components/Nav/NavbarButtonGroup.tsx:
```diff
@@ -0,0 +1,20 @@
+  to: string;
+}
+
+export function NavbarIcon({ text, to }: NavbarButtonGroup) {
```
This name (`NavbarIcon`) can lead to confusion. The component isn't an icon.

PedroChaparro commented on src/components/Nav/NavbarButtonGroup.tsx:
```diff
@@ -0,0 +1,20 @@
+import { Link } from "react-router-dom";
+
+interface NavbarButtonGroup {
```
You have two different names in this component. Is this the `ButtonGroup` or the `NavbarIcon` component?

PedroChaparro commented on src/components/Nav/Nav.tsx:
```diff
@@ -15,8 +22,22 @@ export function NavbarScreen() {
         <div className="flex flex-grow justify-end gap-4">
-          <Button text="Login" to="/login" />
-          <Button text="Register" to="/register" />
+          <div className="lg:hidden">
```
I think you don´t need to wrap the button inside a `div`

PedroChaparro commented on src/components/Nav/Nav.tsx:
```diff
@@ -15,8 +22,22 @@ export function NavbarScreen() {
+            </button>
+          </div>
+          {showMobileMenu && (
+            <div className="absolute left-0 right-0 top-32 flex -translate-y-1/2 transform items-center justify-center bg-white p-2 lg:hidden">
```
1. Use the `md` breakpoint instead of `lg` to hide the buttons. 
2. Try to make a column layout as the shown in the following image. It would be nice if it takes the full screen height: 

![image](https://github.com/hawks-atlanta/frontend-react/assets/62714297/4dde57ad-9742-4ca4-87ca-56fb0cea6aa9)

MiguelMRojas committed:
style: horizontal margin

MiguelMRojas committed:
refactor: rename my component

MiguelMRojas committed:
fix: I removed the div, and corrected the md

MiguelMRojas committed:
ci: test correction in navbar

MiguelMRojas committed:
style: updated buttons

PedroChaparro committed:
chore: Delete unused file

PedroChaparro committed:
refactor: Update component name and styles

PedroChaparro reviewed (changes requested):
Should close the Navbar after navigating to other route. **Pull before implementing the change.**

PedroChaparro commented:
@MiguelMRojas Restore the `Button.spec.cy.tsx` file. It fas empty but apparently the tests fail if there isn't at least one component testing file.

Removed in [f246e4f](https://github.com/hawks-atlanta/frontend-react/pull/21/commits/f246e4f0ba83a53d42ab294b21acc53a661ae61d).

MiguelMRojas committed:
fix: navbar test setup

MiguelMRojas committed:
feat: close menu when changing view

PedroChaparro committed:
test: Update tests selectors

PedroChaparro reviewed (approved):
Squash and merge :shipit:. 

Check my last commit. Using a selector like `md:hidden` isn't a good practice according to the [documentation](https://docs.cypress.io/guides/references/best-practices).
